### ISSUE TITLE
Repo: Flag instances of noFormat that don't do anything

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6128]
### Suggestion

Some test suites, such as https://github.com/typescript-eslint/typescript-eslint/blob/027ce98b8af1a901ee4362cc252295bc03e6ca31/packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts, use `noFormat` because they need some code blocks to preserve formatting. But many of those `noFormat`s end up being copy & pasted across multiple tests, some of which end up not needing it.

Proposal: let's add a lint rule that flags uses of `noFormat` that don't need to be used?

:x: Incorrect:

```ts
noFormat`a.b;`
```

✅ Correct:

```ts
`a.b;`
```

**Comment by bradzacher** [comment]
we could probably include this directly in the [`plugin-test-formatting`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts) rule as a check.

I.e. something like "if code has `noFormat`, format it, and if the output matches the original code - flag the unnecessary `noFormat`"

**Comment by Jolg42** [comment]
Note: I'm preparing a PR for this ✨

**Comment by Jolg42** [comment]
@bradzacher here it is! https://github.com/typescript-eslint/typescript-eslint/pull/6853
It's my first PR here so feel free to add suggestions, this is a bit unfamiliar to me 😄
